spec/requests/graphql/mutations/update_dynamic_field_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Mutations::UpdateDynamicField, type: :request do
+  let(:dynamic_field) { FactoryBot.create(:dynamic_field) }
+
+  include_examples 'requires user to have correct permissions for graphql request' do
+    let(:variables) { { input: { stringKey: dynamic_field.string_key, displayLabel: 'New Location' } } }
+    let(:request) { graphql query, variables }
+  end
+
+  context 'when logged in user is an administrator' do
+    before { sign_in_user as: :administrator }
+
+    context 'when updating display_label' do
+      let(:variables) do
+        {
+          input: {
+            stringKey: dynamic_field.string_key,
+            displayLabel: 'New Location'
+          }
+        }
+      end
+
+      before { graphql query, variables }
+      it 'correctly updates record' do
+        dynamic_field.reload
+        expect(dynamic_field.display_label).to eql 'New Location'
+      end
+    end
+
+    context 'when updating controlled vocabulary' do
+      let(:variables) do
+        {
+          input: {
+            stringKey: dynamic_field.string_key,
+            controlledVocabulary: 'vocabulary2'
+          }
+        }
+      end
+
+      before { graphql query, variables }
+      it 'correctly updates record' do
+        dynamic_field.reload
+        expect(dynamic_field.controlled_vocabulary).to eql 'vocabulary2'
+      end
+    end
+
+    context 'when updating select options' do
+      let(:variables) do
+        {
+          input: {
+            stringKey: dynamic_field.string_key,
+            selectOptions: '{}'
```

Comment:
@JackBlackLight would you mind setting selectOptions to a non-empty hash for this test?  I think it'll make it a more resilient test in case somehow we ever default to an empty hash for a select field.